suite-common/message-system/src/__tests__/experimentUtils.test.ts

```diff
@@ -35,13 +42,27 @@ describe('testing experiment utils', () => {
         expect(isExistInstanceIdNotInVariantRange).toEqual(false);
     });
 
+    const mockRandomInt = (isAGroup: boolean) => {
+        (getWeakRandomId as jest.Mock).mockImplementation(() =>
+            isAGroup ? '1XxK0mjUnx' : 'AyRQxROQKW',
+        );
+    };
+
     it('test selectActiveExperimentGroup share of variant inclusion', () => {
-        const deviation = 0.05;
         const sampleSize = 1000;
         let groupACount = 0;
         let groupBCount = 0;
 
-        const arrayOfIds = getArrayOfInstanceIds(sampleSize);
+        mockRandomInt(true);
+        const arrayOfIdsAGroup = getArrayOfInstanceIds(
+            (sampleSize / 100) * experimentTest.groups[0].percentage,
```

Comment:
Tiniest of nits: rearranging will make it clear _why_ divide by 100: that it's to convert % to fraction, not that we want to "decrease sampleSize"
Same below for :b: 
```suggestion
            sampleSize * (experimentTest.groups[0].percentage / 100),
```